src/SecurityAdvisory/GitHubSecurityAdvisoriesSource.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+
+namespace App\SecurityAdvisory;
+
+use App\Entity\Package;
+use Composer\IO\ConsoleIO;
+use GuzzleHttp\Client;
+
+/**
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class GitHubSecurityAdvisoriesSource implements SecurityAdvisorySourceInterface
+{
+    public const SOURCE_NAME = 'GitHub';
+
+    private Client $guzzle;
+
+    public function __construct(Client $guzzle)
+    {
+        $this->guzzle = $guzzle;
+    }
+
+    public function getAdvisories(ConsoleIO $io, Package $package): ?array
+    {
+        $githubToken = null;
+        $maintainers = $package->getMaintainers();
+        foreach ($maintainers as $maintainer) {
+            if ($maintainer->getGithubToken()) {
+                $githubToken = $maintainer->getGithubToken();
+                break;
+            }
+        }
+
+        if (null === $githubToken) {
+            return [];
+        }
+
+        /** @var RemoteSecurityAdvisory[] $advisories */
+        $advisories = [];
+        $hasNextPage = true;
+        $after = '';
+
+        while ($hasNextPage) {
+            $opts = [
+                'headers' => ['Authorization' => 'Bearer ' . $githubToken],
+                'json' => ['query' => $this->getQuery($package->getName(), $after)],
+            ];
+
+            $response = $this->guzzle->request('POST', 'https://api.github.com/graphql', $opts);
```

Comment:
This should probably catch `GuzzleException` and log the result to avoid having the worker crash